### PR TITLE
Web xr enter exit ui

### PR DESCRIPTION
**Pull request by TrevorDev**
https://github.com/BabylonJS/Babylon.js/issues/3899
Testing PG: http://playground.babylonjs.com#GVAEUW#96

### CODE REVIEW

File: src/Cameras/XR/babylon.webXREnterExitUI.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                        };
+                    }
+                });
+            });
+        }
+        private constructor(private scene: BABYLON.Scene, options: WebXREnterExitUIOptions) {
+            this._overlay = document.createElement("div");
+            this._overlay.style.cssText = "z-index:11;position: absolute; right: 20px;bottom: 50px;";
+
+            var hmdBtn = document.createElement("button");
```
Comment (deltakosh) [inline src/Cameras/XR/babylon.webXREnterExitUI.ts]: we need to test this in the PG and sandbox context to be sure it works everywhere.
Also do you think we could add an option to let the user define the css?

File: src/Cameras/XR/babylon.webXRExperienceHelper.ts [hunk 2]
```diff
@@ -45,27 +54,39 @@ module BABYLON {
         private _nonVRCamera: Nullable<Camera> = null;
         private _originalSceneAutoClear = true;
 
-        private _outputCanvas: HTMLCanvasElement;
-        private _outputCanvasContext: WebGLRenderingContext;
+        private _supported = false;
+
+        /**
+         * Creates the experience helper
+         * @param scene the scene to attach the exprience helper to
```
Comment (deltakosh) [inline src/Cameras/XR/babylon.webXRExperienceHelper.ts]: typo

File: src/Cameras/XR/babylon.webXRManagedOutputCanvas.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+     */
+    export class WebXRManagedOutputCanvas implements IDisposable {
+        private _canvas: Nullable<HTMLCanvasElement> = null;
+        /**
+         * xrpresent context of the canvas which can be used to display/mirror xr content
+         */
+        public canvasContext: Nullable<WebGLRenderingContext> = null;
+        /**
+         * Initializes the canvas to be added/removed upon entering/exiting xr
+         * @param helper the xr expreince helper used to trigger adding/removing of the canvas
```
Comment (deltakosh) [inline src/Cameras/XR/babylon.webXRManagedOutputCanvas.ts]: typo

File: src/Cameras/XR/babylon.webXRManagedOutputCanvas.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+         * @param canvas The canvas to be added/removed (If not specified a full screen canvas will be created)
+         */
+        public constructor(helper: WebXRExperienceHelper, canvas?: HTMLCanvasElement) {
```
Comment (deltakosh) [inline src/Cameras/XR/babylon.webXRManagedOutputCanvas.ts]: I'm wondering why you need to provide a canvas? Could it be jsut an engine? or could we have a canvas which is different from the main engine one?

Comment (TrevorDev) [reply src/Cameras/XR/babylon.webXRManagedOutputCanvas.ts]: This is the canvas that the xr content is displayed on. Initially I wanted it to just use the current canvas that the engine uses but this isn't supported by the current spec (see issue I created https://github.com/immersive-web/webxr/issues/341). So right now I recommend just having a second full screen canvas, which is what this function will create if canvas isn't specified.

File: src/Cameras/XR/babylon.webXREnterExitUI.ts [hunk 5]
```diff
@@ -7,6 +7,7 @@ module BABYLON {
          * Context to enter xr with
          */
         outputCanvasContext?: Nullable<WebGLRenderingContext>;
+        customButtons?: Array<{element: HTMLElement, initializationOptions: XRSessionCreationOptions}>;
```
Comment (deltakosh) [inline src/Cameras/XR/babylon.webXREnterExitUI.ts]: XRSessionCreationOptions or WebXRSessionCreationOptions ?

**Pull request merged**